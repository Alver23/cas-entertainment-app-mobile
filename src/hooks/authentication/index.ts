// Dependencies
import React from 'react';

// Contexts
import { AuthContext } from '@contexts/auth/auth-context';
import { IAuthContext } from '@contexts/auth/auth-context-interface';

export const useAuthentication = () => {
  return React.useContext<IAuthContext>(AuthContext);
};
