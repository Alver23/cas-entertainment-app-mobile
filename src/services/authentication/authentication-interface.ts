// Models
import { IUserAuth } from '@models/user-interface';

export interface ILoginRequest {
  email: string;
  password: string;
}

export type ILoginResponse = IUserAuth;
