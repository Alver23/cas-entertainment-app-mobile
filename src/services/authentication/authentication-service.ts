// Core
import { externalAxiosInstance } from '@core/axios';

// config
import endpoints from '@config/endpoints';

// Connector
import { Connector } from '@core/connector';

// Models
import { ILoginRequest, ILoginResponse } from './authentication-interface';

export class AuthenticationService {
  constructor(private readonly http: Connector) {}

  public login(payload: ILoginRequest): Promise<ILoginResponse> {
    const url = endpoints.login;
    const { email, password } = payload;
    const auth = {
      password,
      username: email,
    };
    return this.http.post(url, {}, { auth });
  }
}

export const authenticationService = new AuthenticationService(new Connector(externalAxiosInstance));
