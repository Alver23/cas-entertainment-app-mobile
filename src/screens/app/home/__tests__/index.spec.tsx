// Dependencies
import React from 'react';
import { shallow, ShallowRendererProps } from 'enzyme';

// Hooks
import * as hooks from '@hooks/authentication';

// Screen
import { HomeScreen } from '../index';

describe('HomeScreen', () => {
  let component: ShallowRendererProps;

  const mockState = {
    logout: jest.fn(),
  };

  beforeEach(() => {
    jest.spyOn(hooks, 'useAuthentication').mockReturnValue(mockState);
    component = shallow(<HomeScreen />);
  });

  it('should save a snapshot of the component', () => {
    expect(component).toMatchSnapshot();
  });
});
