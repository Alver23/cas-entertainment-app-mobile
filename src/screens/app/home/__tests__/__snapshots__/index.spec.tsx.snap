// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppStack should render correctly 1`] = `
<View>
  <Themed.Text>
    HomeScreen
  </Themed.Text>
  <ForwardRef(Themed.Button)
    onPress={[MockFunction]}
    title="Logout"
  />
</View>
`;
